### PR TITLE
feat/78-message-create-input-property-validation closes #78

### LINKED ISSUE #78: Add input property validation for all message `create()` methods
Add input validation for all message `create()` methods by reusing the JSON schema already being used during message processing: https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/validation/validator.ts#L20

Example create method needing update:
https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/interfaces/collections/messages/collections-write.ts#L26

### PR DESCRIPTION
**Pull request by virajjiwane**
Add input property validation for all message create() methods closes #78

### CODE REVIEW

**Review by thehenrytsai — CHANGES_REQUESTED** [review]
@virajjiwane, thanks for working on this and submitting this PR!

The issue is specifically hoping that we can reuse the JSON schema [validation code](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/validation/validator.ts#L20) that we already have so we minimize extra code.

Also `CollectionsWrite` is newly checked in so also needs the same validation.

**Comment by virajjiwane** [comment]
Hi @thehenrytsai, thanks for the reply. I understood the requirement, will work on the changes.

**Comment by virajjiwane** [comment]
@thehenrytsai changes done. please review

File: src/interfaces/collections/messages/collections-write.ts [hunk 1]
```diff
@@ -45,6 +46,9 @@ export class CollectionsWrite extends Message implements Authorizable {
     const authorization = await sign({ descriptor }, options.signatureInput);
     const message = { descriptor, authorization };
 
+    const messageType = descriptor['method'];
```
Comment (thehenrytsai) [inline src/interfaces/collections/messages/collections-write.ts:49]:
```suggestion
    const messageType = descriptor.method;
```

File: src/interfaces/collections/messages/collections-write.ts [hunk 2]
```diff
@@ -45,6 +46,9 @@ export class CollectionsWrite extends Message implements Authorizable {
...
+    validate(messageType, message);
```
Comment (thehenrytsai) [inline src/interfaces/collections/messages/collections-write.ts:50]: validate before `await sign(...)` since signing it costly.

**Review by thehenrytsai — CHANGES_REQUESTED** [review]
Thanks for adding the change, almost there, a couple of comments.

**Comment by virajjiwane** [comment]
@thehenrytsai did changes in all Message create methods and moved validation before any auth functions I could find. please review again.

**Pull request merged**